woocommerce/woocommerce-android
# Feature/8425 launch store screen

Part of: #8425


### Description


This PR implements launch your store task according to design:

![Screenshot 2023-03-10 at 13 06 24](https://user-images.githubusercontent.com/2663464/224312150-765d07bd-1a50-4543-ba10-b5e65ae84616.png)

This PR doesn't include the following: 
- Error handling. Is done here: #8528
- Using the right authenticated web view to be able to show the preview correctly when the store is not launched yet. @0nko was taking a look into that. More context here: pe5sF9-15a-p2#comment-1828
- Refreshing the current plan state after upgrading to the `eCommerce` plan from free trial. That should be part of Eagles's work. 
- Refresh the onboarding list state after store is launched. This should be tackled in a different PR where we add the new refresh policy

### Testing instructions

**eCommerce** plan test
- Create a new store. Preferably using store creation flow from mobile. Disable IAP if needed by updating `IsIAPEnabled` class
- Click on launch store task from onboarding list
- When store launched successfully check the preview is displayed correctly
- Click on share url and check Android share bottom sheet is displayed
- Click on navigate back to store to land on My Store tab again. 

Free trial plan test
- Create a new free trial site by opening this endpoint `https://wordpress.com/setup/wooexpress`
- Click on launch store task
- Check launch task is disable
- Click on the banner and check you are redirected to plans web view for a plan upgrade. 

### Images/gif


Free Trial

https://user-images.githubusercontent.com/2663464/224321176-19180344-517e-46eb-9210-be29bff0062b.mp4


Publish store

https://user-images.githubusercontent.com/2663464/224321202-0b09aae4-c71d-47c3-8bf0-60e9b7d99567.mp4
